resources/lib/kodi/KodiAddon.py

```diff
@@ -36,6 +42,45 @@ class KodiService( KodiAddon ):
 	def __init__( self ):
 		KodiAddon.__init__( self )
 
+class KodiInterlockedMonitor( xbmc.Monitor ):
+	def __init__( self, service, setting_id ):
+		super( KodiInterlockedMonitor, self ).__init__()
+		self.instance_id	= ''.join( format( x, '02x' ) for x in bytearray( os.urandom( 16 ) ) )
+		self.setting_id		= setting_id
+		self.service		= service
+
+	def RegisterInstance( self, waittime = 1 ):
+		if self.BadInstance():
+			self.service.info( 'Found other instance with id {}', self.instance_id )
+			self.service.info( 'Startup delayed by {} second(s) waiting the other instance to shut down', waittime )
+			self.service.setSetting( self.setting_id, self.instance_id )
+			xbmc.Monitor.waitForAbort( self, waittime )
+		else:
+			self.service.setSetting( self.setting_id, self.instance_id )
+
+	def UnregisterInstance( self ):
+		self.service.setSetting( self.setting_id, '' )
+
+	def BadInstance( self ):
+		instance_id = self.service.getSetting( self.setting_id )
+		return instance_id and self.instance_id != instance_id
```

Comment:
Das macht man aber nicht so.
BadInstance return is bad code.

Reply:
Okok! Ich scheibe ja schon `len( instance_id ) > ß`